src/pages/Graph/GraphPage.tsx

```diff
@@ -92,76 +108,140 @@ const GraphErrorBoundaryFallback = () => {
   private pollTimeoutRef?: number;
   private pollPromise?: CancelablePromise<any>;
   private readonly errorBoundaryRef: any;
   private cytoscapeGraphRef: any;
 
+  static getNodeParamsFromProps(props: RouteComponentProps<GraphURLPathProps>): NodeParamsType | undefined {
+    const app = props.match.params.app;
+    const appOk = app && app !== 'unknown' && app !== 'undefined';
+    const namespace = props.match.params.namespace;
+    const namespaceOk = namespace && namespace !== 'unknown' && namespace !== 'undefined';
+    const service = props.match.params.service;
+    const serviceOk = service && service !== 'unknown' && service !== 'undefined';
+    const workload = props.match.params.workload;
+    const workloadOk = workload && workload !== 'unknown' && workload !== 'undefined';
+    if (!appOk && !namespaceOk && !serviceOk && !workloadOk) {
+      return;
+    }
+
+    let nodeType;
+    let version;
+    if (appOk || workloadOk) {
+      nodeType = appOk ? NodeType.APP : NodeType.WORKLOAD;
+      version = props.match.params.version;
+    } else {
+      nodeType = NodeType.SERVICE;
+      version = '';
+    }
+    const node: NodeParamsType = {
+      app: app,
+      namespace: { name: namespace },
+      nodeType: nodeType,
+      service: service,
+      version: version,
+      workload: workload
+    };
+    return node;
+  }
+
+  static isNodeChanged(prevNode?: NodeParamsType, node?: NodeParamsType): boolean {
+    if (prevNode === node) {
+      return false;
+    }
+    if ((prevNode && !node) || (!prevNode && node)) {
+      return true;
+    }
+    if (prevNode && node) {
+      const nodeAppHasChanged = prevNode.app !== node.app;
+      const nodeServiceHasChanged = prevNode.service !== node.service;
+      const nodeVersionHasChanged = prevNode.version !== node.version;
+      const nodeTypeHasChanged = prevNode.nodeType !== node.nodeType;
+      const nodeWorkloadHasChanged = prevNode.workload !== node.workload;
+      return (
+        nodeAppHasChanged ||
+        nodeServiceHasChanged ||
+        nodeVersionHasChanged ||
+        nodeWorkloadHasChanged ||
+        nodeTypeHasChanged
+      );
+    }
+    return false;
+  }
+
   constructor(props: GraphPageProps) {
     super(props);
     this.errorBoundaryRef = React.createRef();
     this.cytoscapeGraphRef = React.createRef();
     this.state = {
       showHelp: false
     };
+
+    // Let URL override current redux state at construction time
+    // Note that state updates will not be posted until until after the first render
+    const urlNode = GraphPage.getNodeParamsFromProps(props);
+    if (GraphPage.isNodeChanged(urlNode, props.node)) {
+      props.setNode(urlNode);
+    }
   }
 
   componentDidMount() {
-    this.scheduleNextPollingInterval(0);
+    // This is a special bookmarking case. If the initial URL is for a node graph then
+    // defer the graph fetch until the first component update, when the node is set.
+    if (!store.getState().graph.node) {
```

Comment:
Could we get the node from the url instead of the store?

Reply:
I'm going to say yes, I think so, if we want to do the URL parsing a second time.  Why do you ask, to avoid direct access of the store?

Reply:
Yes, to avoid direct access to the store.

Reply:
We can make a change in the future if you like.